### PR TITLE
Send SDL_APP_TERMINATING before unload (#88)

### PR DESCRIPTION
**Pull request by Beuc**

### CODE REVIEW

File: src/video/emscripten/SDL_emscriptenevents.c [hunk 1]
```diff
@@ -652,6 +652,16 @@ Emscripten_HandleVisibilityChange(int eventType, const EmscriptenVisibilityChang
     return 0;
 }
 
+static EM_BOOL
+Emscripten_HandleBeforeUnload(int eventType, const void* reserved, void *userData)
```
Comment (Daft-Freak) [inline src/video/emscripten/SDL_emscriptenevents.c]: `void *reserved`

**Comment by Daft-Freak** [comment]
Looks good otherwise, though not sure we need the comment about using `SDL_AddEventWatch` as the `SDL_APP_*` events always need to be handled like that.

**Comment by Beuc** [comment]
Ren'Py for instance seems to get away with handling SDL_APP_* in the main thread on Android & iOS, so I guess it's worth making clear this callback is handled differently than the others (to the least my future self will be grateful if I have to work on this code again ;)).
Also I fixed the return value with didn't match the callback signature.

**Comment by Beuc** [comment]
Fun fact: in both Firefox and Chromium the _main loop_ may receive `SDL_APP_TERMINATING`.
Similarly FS.sync is not guaranteed to complete (as it's asynchronous, sadly).
Typically if I spam the click button and reload meanwhile, though in practice this is rare.
It seems the browser still process a random number of events before finalizing the unload. Only the SDL_AddEventWatch callback itself is guaranteed to run.

**Comment by Beuc** [comment]
Let me know if you need something else :)

**Comment by Daft-Freak** [comment]
Thanks. (Didn't notice the different return type myself...)

**Pull request merged**